control-plane/plugin/src/resources/node.rs

```diff
@@ -211,24 +258,74 @@ impl CreateRows for NodeDisplay {
         match self.format {
             NodeDisplayFormat::Default => self.inner.create_rows(),
             NodeDisplayFormat::CordonLabels => {
-                let mut rows = self.inner.create_rows();
-                let cordon_labels_string = self
-                    .inner
-                    .spec
-                    .as_ref()
-                    .unwrap_or(&NodeSpec::default())
-                    .cordon_labels
-                    .join(", ");
-
-                // Add the cordon labels to each row.
-                rows.iter_mut()
-                    .for_each(|row| row.add_cell(Cell::new(&cordon_labels_string)));
+                let mut rows = vec![];
+                for node in self.inner.iter() {
+                    let mut row = node.create_rows();
+                    let labelstring = NodeDisplay::get_label_list(node).join(", ");
+                    // Add the cordon labels to each row.
+                    row[0].add_cell(Cell::new(&labelstring));
+                    rows.push(row[0].clone());
+                }
+                rows
+            }
+            NodeDisplayFormat::Drain => {
+                let mut rows = vec![];
+                for node in self.inner.iter() {
+                    let mut row = node.create_rows();
+
+                    let drain_status_string = match &node.spec.as_ref().unwrap().cordondrainstate {
+                        Some(ds) => match ds {
+                            CordonDrainState::cordonedstate(_) => "Not draining",
+                            CordonDrainState::drainingstate(_) => "Draining",
+                            CordonDrainState::drainedstate(_) => "Drained",
+                        },
+                        None => "Not draining",
+                    };
+
+                    let labelstring = NodeDisplay::get_drain_label_list(node).join(", ");
+                    // Add the drain labels to each row.
+                    row[0].add_cell(Cell::new(drain_status_string));
+                    row[0].add_cell(Cell::new(&labelstring));
+                    rows.push(row[0].clone());
+                }
                 rows
             }
         }
     }
 }
 
+pub fn node_display_print(
```

Comment:
Can we make this `pub(crate)` also doc comments are needed.

Reply:
fixed